pylint/checkers/refactoring/refactoring_checker.py

```diff
@@ -656,17 +662,32 @@ def leave_module(self, _: nodes.Module) -> None:
         )
         self._init()
 
-    @utils.only_required_for_messages("too-many-nested-blocks", "no-else-return")
+    @utils.only_required_for_messages(
+        "too-many-nested-blocks", "no-else-return", "return-or-yield-in-finally"
+    )
     def visit_tryexcept(self, node: nodes.TryExcept | nodes.TryFinally) -> None:
         self._check_nested_blocks(node)
 
         if isinstance(node, nodes.TryExcept):
             self._check_superfluous_else_return(node)
             self._check_superfluous_else_raise(node)
 
+        if isinstance(node, nodes.TryFinally):
+            for final_node in node.finalbody:
+                self._check_yield_return_in_finally(final_node)
+
     visit_tryfinally = visit_tryexcept
     visit_while = visit_tryexcept
 
+    def _check_yield_return_in_finally(self, node: nodes.NodeNG) -> None:
+        if isinstance(node, nodes.Return):
+            self.add_message("return-or-yield-in-finally", node=node, args=("return",))
+        if isinstance(node, nodes.Yield):
+            self.add_message("return-or-yield-in-finally", node=node, args=("yield",))
+
+        for child in node.get_children():
+            self._check_yield_return_in_finally(child)
```

Comment:
It's possible to use ``for child_node in node.nodes_of_class(nodes.Yield, nodes.Return)`` to recover only the yield/return nodes in a generator.

Reply:
Oh my, that is good to know. Thanks!